src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -81,8 +81,8 @@ public boolean equals(Object o) {
         }
 
         ViewCommand other = (ViewCommand) o;
-        return targetName.equals(other.targetName)
-                || targetIndex.equals(other.targetIndex); // state check
+        return (targetName != null && targetName.equals(other.targetName))
+                || (targetIndex != null && targetIndex.equals(other.targetIndex)); // state check
```

Comment:
okay will keep this in mind for the other .equals methods that uses name/index flags.